src/awkward/contents/bitmaskedarray.py

```diff
@@ -632,7 +632,7 @@ def _completely_flatten(self, backend, options):
         if branch or options["drop_nones"] or depth > 1:
             return self.project()._completely_flatten(backend, options)
         else:
-            return [self.simplify_optiontype()]
+            return [self]
```

Comment:
Here's another unbalanced `simplify_optiontype` → `simplified` pair, but in the future, `self` can't be non-canonical.